config/deploy/xp5k.rb

```diff
@@ -3,39 +3,42 @@
 require 'xp5k'
 require 'erb'
 
-# G5K global parameters
-set :site, ENV['site'] || "lyon"
-set :walltime, ENV['walltime'] || "3:00:00"
-set :subnet, ENV['subnet'] || "slash_18"
-set :jobname, ENV['jobname'] || "openstack"
-## vlantype values:
-# "kavlan"        - routed local vlan
-# "kavlan-global" - global vlan
-set :vlantype, ENV['vlantype'] || "kavlan"
-
+# Load ./xp.conf file
+#
 XP5K::Config.load
 
+# Defaults configuration
+#
+XP5K::Config[:jobname]    ||= 'openstack'
+XP5K::Config[:site]       ||= 'toulouse'
+XP5K::Config[:walltime]   ||= '1:00:00'
+XP5K::Config[:cluster]    ||= ''
+XP5K::Config[:vlantype]   ||= 'kavlan'
+XP5K::Config[:nodes]      ||= '3'
+
+cluster = "and cluster='" + XP5K::Config[:cluster] + "'" if !XP5K::Config[:cluster].nil? || !XP5K::Config[:cluster].empty?
```

Comment:
I would have put 
```ruby
if !XP5K::Config[:cluster].nil? && !XP5K::Config[:cluster].empty?
```
or 
```ruby
unless XP5K::Config[:cluster].nil? || XP5K::Config[:cluster].empty?
```

or more simply as we set the empty string just above

```ruby
if !XP5K::Config[:cluster].empty?
```